pepper-apis/src/main/java/org/broadinstitute/ddp/service/PdfGenerationService.java

```diff
@@ -720,6 +707,128 @@ private void convertSubstitutionToPdf(AnswerSubstitution substitution,
         }
     }
 
+    private void substituteComposite(CompositeAnswerSubstitution substitution, PdfAcroForm form, CustomTemplate template,
+                                     List<String> errors, Answer answer) throws IOException {
+        if (answer == null) {
+            return;
+        }
+
+        //passed form contains custom composite template pages. just remove those empty pages
+        int pageCount = form.getPdfDocument().getNumberOfPages();
+        IntStream.range(1, pageCount + 1).forEach(i -> form.getPdfDocument().removePage(i));
```

Comment:
Not understanding what this block of code does. Why are they empty? Why do we need to remove them?

Reply:
Each composite question is one pdf page.  Form already includes empty composite template/page (existing code). Code in this method does not use that already included page. Iterates through answers and creates 1 or more pages depending on how many answers were provided (List of Answers), hence deleting those empty pages.

Reply:
I see. So the caller, `generateCustomPdf()`, had initialized the `form` with the template and we're clearing that out before adding in rendered composite pages.